AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -158,11 +163,55 @@ + (void)deleteDocumentWithPartition:(NSString *)partition
   (void)completionHandler;
 }
 
-#if TARGET_OS_OSX
-- (void)dealloc {
-}
+#pragma mark - MSDataStore Implementation
+- (void)createWithPartition:(NSString *)partition
+                 documentId:(NSString *)documentId
+                   document:(id<MSSerializableDocument>)document
+               writeOptions:(MSWriteOptions *)__unused writeOptions
+          completionHandler:(MSDocumentWrapperCompletionHandler)completionHandler {
 
-#endif
+  [MSTokenExchange tokenAsync:(MSStorageIngestion *)self.ingestion
+                   partitions:@[ partition ]
+            completionHandler:^(MSTokensResponse *_Nonnull tokenResponses, NSError *_Nonnull error) {
+
+              // If error getting token.
+              if (error && !tokenResponses) {
+                MSLogError([MSDataStore logTag], @"Can't get CosmosDb token: %@", [error description]);
+                completionHandler([[MSDocumentWrapper alloc] initWithError:error documetnId:documentId]);
+              }
+
+              // Create http client.
+              MSCosmosDbIngestion *cosmosDbIngestion = [[MSCosmosDbIngestion alloc] init];
+
+              NSError *serializationError;
+              NSData *body = [NSJSONSerialization dataWithJSONObject:[document serializeToDictionary] options:0 error:&serializationError];
+              if (!body && serializationError) {
```

Comment:
Doesn't allow any docs without contents? Like 0 byte size file.
Also, `&&` could be `||`.